service/src/test/org/apache/hive/service/server/TestHS2HttpServerLDAP.java

```diff
@@ -0,0 +1,197 @@
+import org.apache.hive.service.auth.HttpAuthUtils;
+import org.apache.hive.service.auth.PasswdAuthenticationProvider;
+import org.apache.hive.service.auth.ldap.LdapAuthService;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.CookieStore;
+import org.apache.http.cookie.Cookie;
+import org.apache.http.impl.client.BasicCookieStore;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.params.BasicHttpParams;
+import org.apache.http.params.HttpParams;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.util.B64Code;
+import org.eclipse.jetty.util.StringUtil;
+import java.util.List;
+import java.util.Optional;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import javax.security.sasl.AuthenticationException;
+
+public class TestHS2HttpServerLDAP {
+
+  private static HiveServer2 hiveServer2;
+  private static Integer webUIPort;
+  private static final String HOST = "localhost";
+  private static String metastorePasswd = "693efe9fa425ad21886d73a0fa3fbc70"; //random md5
+  private static final String VALID_USER = "validUser";
+  private static final String VALID_PASS = "validPass";
+  private static final String INVALID_USER = "invalidUser";
+  private static final String INVALID_PASS = "invalidPass";
+
+  @BeforeClass
+  public static void beforeTests() throws Exception {
+    webUIPort =
+        MetaStoreTestUtils.findFreePortExcepting(Integer.parseInt(ConfVars.HIVE_SERVER2_WEBUI_PORT.getDefaultValue()));
+    HiveConf hiveConf = new HiveConf();
+    hiveConf.setBoolVar(ConfVars.HIVE_IN_TEST, true);
+    hiveConf.set(ConfVars.HIVE_SERVER2_WEBUI_PORT.varname, webUIPort.toString());
+    hiveConf.setBoolVar(ConfVars.HIVE_SERVER2_WEBUI_ENABLE_LDAP, true);
+    hiveConf.set(ConfVars.METASTORE_PWD.varname, metastorePasswd);
+    hiveConf.setVar(HiveConf.ConfVars.HIVE_AUTHORIZATION_MANAGER,
+        "org.apache.hadoop.hive.ql.security.authorization.plugin.sqlstd.SQLStdHiveAuthorizerFactory");
+    PasswdAuthenticationProvider authenticationProvider = new DummyLdapAuthenticationProviderImpl();
+    hiveServer2 = new HiveServer2(authenticationProvider);
+    hiveServer2.init(hiveConf);
+    hiveServer2.start();
+    Thread.sleep(5000);
+  }
+
+  @Test
+  public void testValidCredentialsWithAuthorizationHeader() throws Exception {
+    CloseableHttpClient httpclient = null;
+    try {
+      CookieStore httpCookieStore = new BasicCookieStore();
+      HttpClientBuilder builder = HttpClientBuilder.create().setDefaultCookieStore(httpCookieStore);
+      httpclient = builder.build();
+
+      HttpGet httpGet = new HttpGet("http://" + HOST + ":" + webUIPort + "/jmx");
+      String authB64Code = B64Code.encode(VALID_USER + ":" + VALID_PASS, StringUtil.__ISO_8859_1);
```

Comment:
using __ISO_8859_1 looks strange, is there any specific reason?

Reply:
I did in the same way as in existing examples in Hive code. Unfortunately, I don't know what is the reason behind setting this encoding when encoding basic auth credentials, but this is done this way in several classes in Hive.
https://github.com/apache/hive/blob/e0bd9eac32851b27976a78cd83584cb867b83182/service/src/test/org/apache/hive/service/server/TestHS2HttpServerPam.java#L89-L91

Reply:
okay, it's really a problem, we can get rid of those in a different patch